### DISCUSSION TITLE
How to exclude folders from scan?

### DISCUSSION THREAD

**Discussion by andyshinn** [Checkmarx/kics#5365] [Q&A] [answered]
How can I exclude the `.terraform` folder from my scans? I see there is a `-e` switch to the `scan` command. But I have tried several permutations that don't seem to do anything:

- `-e ".terraform"`
- `-e ".terraform/*"`
- `-e "*/.terraform"`
- `-e "*/.terraform/*"`
- `-e "**/.terraform"`

But the `.terraform` folder in each module still gets scanned. How do I exclude it?

**Comment by rafaela-soares** [comment, marked as answer]
Hello, @andyshinn! Thank you so much for reaching us!

Are you using Docker? If yes, you need to point the volume like this:

-  if `.terraform` is inside of the root of `/path/`:

`docker run -v $pwd/:/path/ checkmarx/kics:latest scan -p /path/ --no-progress -e "/path/.terraform" -v`

- if  `.terraform` is inside of the root of `/path/folder1`:

`docker run -v $pwd/:/path/ checkmarx/kics:latest scan -p /path/ --no-progress -e "/path/*/.terraform" -v`

Let me know if it helps, please 😊

**Reply by andyshinn** [reply]
It does seem to work if the path is exact to the folder. But what if I have many `.terraform` folders in the path from many modules? Is there a way to exclude any `.terraform` folder regardless of depth?

**Reply by rafaela-soares** [reply]
KICS uses the function `Glob` from `path/filepath`, which seems to have limitations for your use case. We will open an issue and take care of it as soon as possible. Thank you so much for noticing and reporting, @andyshinn!

**Reply by andyshinn** [reply]
I saw that globbing was released. But I still cannot get it to work. I tried all the options in my original issue and it does not exclude any `.terraform` folder. How can I exclude all `.terraform` folders?

**Reply by rafaela-soares** [reply]
Hello, @andyshinn

Did you try: `docker run -v $pwd/:/path/ checkmarx/kics:latest scan -p /path/ --no-progress -e "/path/**/.terraform" -v`?

**Reply by andyshinn** [reply]
I will try with Docker. Does it need an absolute path? I was trying with the binary built on macOS. I was expecting it to use relative paths.

**Reply by rafaela-soares** [reply]
@andyshinn, you can use your relative path `-e <your-relative-project-path>/**/.terraform` or `-e ./**/.terraform`

**Reply by andyshinn** [reply]
I still cannot get this to work. If I run `docker run -t -v $PWD/terraform/vht-organization:/path checkmarx/kics:latest scan -p /path -o "/path/" -e /path/**/.terraform` or `docker run -t -v $PWD/terraform/vht-organization:/path checkmarx/kics:latest scan -p /path -o "/path/" -e ./**/.terraform` it is scanning modules and file sin `.terraform` directories. What am I missing?

**Comment by kaplanlior** [comment]
Tagging #5369 as it's easy to miss the reference by GitHub on the right side menu.
